### ISSUE TITLE
How to extract attention maps (spatial & temporal) from flex_attention?

### ISSUE THREAD

**Issue by archernsy** [svg-project/Sparse-VideoGen#23]
Hi, thank you for your excellent work! 🙌

I’d like to ask how to extract the spatial and temporal attention maps from flex_attention, since it only returns out (hidden states). Thanks!

**Comment by haochengxi** [comment]
You can print the flex_attention's attention map by printing it: `print(block_mask)`. The attention mask should looks like this:
<img src="https://github.com/user-attachments/assets/d15ec8d1-57eb-4c0b-a6d5-7e2a981d774f" width="300">


You can also get the simulation of our attention mask here:
https://github.com/svg-project/Sparse-VideoGen/blob/3c71be2d5a6d5adbcfe6d4ad92deff5081eb1954/hyvideo_inference.py#L77
https://github.com/svg-project/Sparse-VideoGen/blob/3c71be2d5a6d5adbcfe6d4ad92deff5081eb1954/hyvideo_inference.py#L125

**Comment by archernsy** [comment]
Thank you for your reply! 
Sorry, I think I didn’t phrase my previous question clearly.
I’m actually trying to get the QK similarity matrices from flex_attention, like the ones shown in Figure 3(b) and 3(d) of your paper. Is there a way to access those?

**Comment by haochengxi** [comment]
If you want to get the softmax(QK^T) matrix, you can directly calculate them head by head (since the attention map for all heads is too large and can not fit in the GPU memory). In this case, you should not use flex_attention, and should calculate the attention score directly.
